lib/FusionInventory/Agent/Task/Inventory/Virtualization/Docker.pm

```diff
@@ -0,0 +1,123 @@
+package FusionInventory::Agent::Task::Inventory::Virtualization::Docker;
+
+use strict;
+use warnings;
+
+use JSON::PP;
+
+use FusionInventory::Agent::Tools;
+
+sub isEnabled {
+    return canRun('docker');
+}
+
+sub doInventory {
+    my (%params) = @_;
+
+    my $inventory = $params{inventory};
+    my $logger    = $params{logger};
+
+    foreach my $container (_getContainers(
+        logger => $logger,
+        command => 'docker ps -a'
+    )) {
+        $inventory->addEntry(
+            section => 'VIRTUALMACHINES', entry => $container
+        );
+    }
+}
+
+sub  _getContainers {
+    my (%params) = @_;
+
+    my $handle = getFileHandle(%params);
+
+    return unless $handle;
+
+    my $entete = <$handle>;
+    chomp $entete;
+    my @entete = split /  +/, $entete;
+    # find PORTS index in header
+    # it's the only one value that can be empty
+    my $portsIndex;
+    my $j = 0;
+    my %entete = map { $_ => $j++ } @entete;
+    if (defined $entete{PORTS}) {
+        $portsIndex = $entete{PORTS};
+    }
+
+    my @containers;
+    while (my $line = <$handle>) {
+        chomp $line;
+        my @info = split(/  +/, $line);
```

Comment:
Hum. Yes, it's right. I have to change this split now. The good thing is that the command is always placed between two '"' (double quotes).